### PR TITLE
Code cleanup and formatting for System.Memory files.

### PR DESCRIPTION
**Pull request by ahsonkhan**
I tried to limit the changes mainly to files shared between coreclr/corefx relevant to System.Memory.

Each set of changes is a separate commit.

The IDE00XX errors were fixed by VS, so all instances of the errors within the solution got fixed.

Related PR on the corefx side: https://github.com/dotnet/corefx/pull/28875

cc @dotnet/corefxlab-contrib, @stephentoub, @jkotas 

This should get mirrored - cc @Anipik, @safern

### CODE REVIEW

File: src/mscorlib/shared/System/Lazy.cs [hunk 1]
```diff
@@ -383,7 +383,7 @@ private void PublicationOnlyViaFactory(LazyHelper initializer)
         private void PublicationOnlyWaitForOtherThreadToPublish()
         {
             var spinWait = new SpinWait();
-            while (!ReferenceEquals(_state, null))
+            while (!(_state is null))
```
Comment (jkotas) [inline src/mscorlib/shared/System/Lazy.cs]: The other places that check `_state` are still using ReferenceEquals. Using `is` here while keeping the rest on the ReferenceEquals plan does not look like an improvement.

Comment (jkotas) [inline src/mscorlib/shared/System/Lazy.cs]: Why is this not `!= null` like the other place?

**Comment by ahsonkhan** [comment]
@dotnet-bot test Windows_NT x64 Checked corefx_baseline
@dotnet-bot test Ubuntu x64 Checked corefx_baseline

File: src/mscorlib/shared/System/Lazy.cs (hunk 1 above)
Comment (ahsonkhan) [reply src/mscorlib/shared/System/Lazy.cs]: For comparing an object to a non-null object, is there any benefit of using ReferenceEquals over `==`? If not, I can change the other calls as well.

Comment (ahsonkhan) [reply src/mscorlib/shared/System/Lazy.cs]: VS defaulted to this as part of the "Null check can be simplified" fix. I will update to use `!=`

File: src/mscorlib/src/System/Reflection/Emit/ModuleBuilder.cs [hunk 2]
```diff
@@ -1139,15 +1139,15 @@ private TypeToken GetTypeTokenWorkerNoLock(Type type, bool getGenericDefinition)
...
+                return new TypeToken(GetTypeRefNested(type, this, string.Empty));
             }
 
             // After this point, the referenced module is not the same as the referencing
             // module.
             //
             ModuleBuilder refedModuleBuilder = refedModule as ModuleBuilder;
 
-            String strRefedModuleFileName = String.Empty;
+            String strRefedModuleFileName = string.Empty;
```
[the remark this answers is not in the document]
Comment (ahsonkhan) [reply src/mscorlib/src/System/Reflection/Emit/ModuleBuilder.cs]: It looks like VS didn't catch all instances. Will fix.

File: src/mscorlib/shared/System/Lazy.cs (hunk 1 above)
Comment (jkotas) [reply src/mscorlib/shared/System/Lazy.cs]: `==` may use overriden operator Equals if there is one.

`ReferenceEquals` is explicit about your intent: that you want to compare references, and not use any overridden operators.

**Comment by joshfree** [comment]
As this change isn't required to ship 2.1 RC, please hold off on merging this until the master branch opens for 2.2 PRs later this month

File: src/mscorlib/shared/System/Memory.cs [hunk 3]
```diff
@@ -6,8 +6,7 @@
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-using EditorBrowsableAttribute = System.ComponentModel.EditorBrowsableAttribute;
-using EditorBrowsableState = System.ComponentModel.EditorBrowsableState;
+using System.ComponentModel;
```
Comment (AtsushiKan) [inline src/mscorlib/shared/System/Memory.cs]: I preferred it the original way. `System.ComponentModel` is not an obvious namespace to import into a file like `Memory.cs` - spelling it out the types makes it clear why we're importing it.

**Pull request merged**